exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlConverter.java

```diff
@@ -481,6 +485,19 @@ public RelOptTableImpl getTable(final List<String> names) {
             .message("Temporary tables usage is disallowed. Used temporary table name: %s.", names)
             .build(logger);
       }
+
+      // Check the schema and throw a valid SchemaNotFound exception instead of TableNotFound exception.
```

Comment:
Does it mean that Calcite instead of returning schema not found exception returns table not found exception?
Per my understanding this PR customizes Drill but what if we go different path and enhance Calcite (or may be this is already done in newer Calcite versions)?

Reply:
Thank you for the review. I agree that this should ideally be handled at Calcite layer. I also think that even after Calcite providing this functionality there should be some customization that needs to be done as we understand the context better than Calcite. Once the calcite fixes this issue then we can always change the code accordingly.

Reply:
Could you please factor out this logic in a separate method?